packages/teams-js/test/internal/videoPerformanceStatistics.spec.ts

```diff
@@ -0,0 +1,131 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable strict-null-checks/all */
+import { VideoFrameTick } from '../../src/internal/videoFrameTick';
+import {
+  VideoPerformanceStatistics,
+  VideoPerformanceStatisticsResult,
+} from '../../src/internal/videoPerformanceStatistics';
+
+jest.useFakeTimers();
+function advanceTimersByTime(time: number): void {
+  jest.advanceTimersByTime(time);
+  VideoFrameTick.tick();
+}
+
+describe('PerformanceStatistics', () => {
+  let result: VideoPerformanceStatisticsResult | undefined;
+  let performanceStatistics: VideoPerformanceStatistics;
+  let reportFunc = jest.fn();
+  reportFunc.mockImplementation((r) => (result = r));
+
+  beforeEach(() => {
+    result = undefined;
+    reportFunc = jest.fn();
+    reportFunc.mockImplementation((r) => (result = r));
+    performanceStatistics = new VideoPerformanceStatistics(1000, reportFunc);
+  });
+  afterEach(() => {
+    jest.clearAllMocks();
+    jest.clearAllTimers();
+  });
+
+  it('reports statistics on timeout', () => {
+    performanceStatistics.processStarts('effectId', 100, 100);
+    performanceStatistics.processEnds();
+    advanceTimersByTime(1000);
+    expect(result).toEqual({
+      effectId: 'effectId',
+      frameHeight: 100,
+      frameWidth: 100,
+      duration: expect.any(Number),
+      sampleCount: 1,
+      distributionBins: expect.anything(),
+    });
+    expect(result?.distributionBins[0]).toEqual(1);
+  });
+
+  it('reports statistics on effect change', () => {
+    performanceStatistics.processStarts('effectId1', 100, 100);
+    advanceTimersByTime(10);
+    performanceStatistics.processEnds();
+    performanceStatistics.processStarts('effectId2', 100, 100);
+    advanceTimersByTime(20);
+    performanceStatistics.processEnds();
+    expect(result).toEqual({
+      effectId: 'effectId1',
+      frameHeight: 100,
+      frameWidth: 100,
+      duration: 10,
+      sampleCount: 1,
+      distributionBins: expect.anything(),
+    });
+    expect(result?.distributionBins[10]).toEqual(1);
+  });
+
+  it('reports statistics on frame size change', () => {
+    performanceStatistics.processStarts('effectId', 100, 100);
+    advanceTimersByTime(10);
+    performanceStatistics.processEnds();
+    performanceStatistics.processStarts('effectId', 200, 200);
+    advanceTimersByTime(20);
+    performanceStatistics.processEnds();
+    expect(result).toEqual({
+      effectId: 'effectId',
+      frameHeight: 100,
+      frameWidth: 100,
+      duration: 10,
+      sampleCount: 1,
+      distributionBins: expect.anything(),
+    });
+    expect(result?.distributionBins[10]).toEqual(1);
+  });
+
+  it("don't report statistics when session is not created", () => {
```

Comment:
I would have expected this to cause a lint error...weird!

```suggestion
  it('don't report statistics when session is not created', () => {
```

Reply:
Prettier automatically format these string, I guess it is because there is a `'` in "don*'*t"